near-sdk-macros/src/lib.rs

```diff
@@ -116,6 +117,7 @@ pub fn near_bindgen(attr: TokenStream, item: TokenStream) -> TokenStream {
 
     if let Ok(input) = syn::parse::<ItemStruct>(item.clone()) {
         let metadata = core_impl::contract_source_metadata_const(attr);
+        let metadata_impl_gen = generate_metadata_impl(&input.ident, &input.generics);
```

Comment:
This way the generated function will only be implemented on the contract struct, but not exposed as contract methods and ABI is not generated for it. This has to be fixed and also test added as we obviously could have missed it unless I happened to spot it.